src/EditorFeatures/CSharpTest/UseObjectInitializer/UseObjectInitializerTests.cs

```diff
@@ -576,6 +576,106 @@ public void Main()
 
         int horse = 1;
     }
+}");
+        }
+
+        [WorkItem(23368, "https://github.com/dotnet/roslyn/issues/23368")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)]
+        public async Task TestWithExplicitImplementedInterfaceMembers1()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+interface IExample {
+    string Name { get; set; }
+}
+
+class C : IExample {
+    string IExample.Name { get; set; }
+}
+
+class MyClass
+{
+    public void Main()
+    {
+        IExample e = [||]new C();
+        e.Name = string.Empty;
+    }
+}");
+        }
+
+        [WorkItem(23368, "https://github.com/dotnet/roslyn/issues/23368")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)]
+        public async Task TestWithExplicitImplementedInterfaceMembers2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"
+interface IExample {
+    string Name { get; set; }
+    string LastName { get; set; }
+}
+
+class C : IExample {
+    string IExample.Name { get; set; }
+    public string LastName { get; set; }
+}
+
+class MyClass
+{
+    public void Main()
+    {
+        IExample e = [||]new C();
+        e.Name = string.Empty;
```

Comment:
so the order causing one to work and not the other? would it be possible to support both if possible?